pyoptsparse/pyParOpt/ParOpt.py

```diff
@@ -223,32 +240,20 @@ def evalObjConGradient(self, x, g, A):
 
                 # Create the quasi-Newton Hessian approximation
                 qn = _ParOpt.LBFGS(problem, subspace=qn_subspace_size)
+                subproblem = _ParOpt.QuadraticSubproblem(problem, qn)
 
                 # Create the trust region problem
-                tr = _ParOpt.TrustRegion(problem, qn, tr_init_size, tr_min_size, tr_max_size, tr_eta, tr_penalty_gamma)
+                tr = _ParOpt.TrustRegion(subproblem, tr_init_size,
+                                         tr_min_size, tr_max_size,
+                                         tr_eta, tr_penalty_gamma)
 
                 # Create the ParOpt problem
-                opt = _ParOpt.InteriorPoint(tr, qn_subspace_size, _ParOpt.NO_HESSIAN_APPROX)
+                opt = _ParOpt.InteriorPoint(subproblem, qn_subspace_size,
+                                            _ParOpt.NO_HESSIAN_APPROX)
 
                 # Set the ParOpt options
                 self._set_paropt_options(opt)
 
-                # Set the output file name
-                opt.setOutputFile(filename)
-
-                # Set the penalty parameter internally in the
-                # code. These must be consistent between the trust
-                # region object and ParOpt.
-                opt.setPenaltyGamma(tr_penalty_gamma)
-
-                # Set parameters for ParOpt in the subproblem
-                opt.setMaxMajorIterations(tr_max_iterations)
-                opt.setAbsOptimalityTol(tr_opt_abs_tol)
```

Comment:
@gjkennedy does this function call no longer exist in the new version of ParOpt? `flake8` is complaining that `tr_opt_abs_tol` is now no longer used.

Reply:
opt.setAbsOptimalityTol(tr_opt_abs_tol) should be called after set_paropt_options so that the tolerance that the trust region subproblems are solved to can be set independently from the overall optimization problem.

Reply:
Right, there is no `opt.setAbsOptimalityTol(tr_opt_abs_tol)` call at the moment.